Yank/repex.py

```diff
@@ -1413,19 +1440,16 @@ def _write_dict(self, path, data, storage_name='analysis',
 # REPLICA-EXCHANGE SIMULATION
 # ==============================================================================
 
-class ReplicaExchange(object):
-    """Replica-exchange simulation facility.
-
-    This base class provides a general replica-exchange simulation facility,
-    allowing any set of thermodynamic states to be specified, along with a
-    set of initial positions to be assigned to the replicas in a round-robin
-    fashion.
+class MultiStateSampler(object):
```

Comment:
Is this class ever intended to be used by itself? Could it be used by itself? The way its structured now it looks like a parallel sampler with no sampler mixing scheme of any kind. Which is fine of course, but if thats the case, I would like to see some tests for it.

If we never intend for it to be used alone, having a way to abstract this, such as the `default_title` or something would be helpful.

Reply:
Yes, it can and will be used by itself. It's a parallel sampler with no mixing scheme.

You're right---there are a few dummy methods that are overridden by `ReplicaExchange` that can use some tests. Instead of duplicating the entire 700-line `TestReplicaExchange` class, is there a way we can refactor that test so we can test both classes in the same way? Perhaps I could rename it `TestMultiStateSampler` and subclass it to create `TestReplicaExchange`, overriding only a couple of the methods as needed? Would that work?

Reply:
You could have `TestMultiStateSampler` made of mostly just the `MultiStateSampler` tests and then a few unique `ReplicaExchangeSampler` tests. I think if you subclassed the `TestMultiStateSampler` it would then re-run all the tests again. Which in the complete case is fine, but we do have a limited length of test time on Travis to consider (we are not close though). I think as a first pass your idea is good, we can refine it later